### DISCUSSION TITLE
What is difference in quarto render and quarto publish in quarto action?

### DISCUSSION THREAD

**Discussion by Pratichhya** [quarto-dev/quarto-cli#5085] [Q&A]
What is difference in quarto render and quarto publish in quarto action?

**Comment by cderv** [comment]
* [`quarto-actions/render`](https://github.com/quarto-dev/quarto-actions/tree/main/render) will only render the project inside the github action runner. It is basically doing `quarto render`
* [`quarto-actions/publish`](https://github.com/quarto-dev/quarto-actions/tree/main/publish) will render and publish - it is running `quarto publish`


Using the former is useful when you want to 
* Publish to another service not supported by `quarto publish` - See this [example](https://github.com/quarto-dev/quarto-actions/blob/main/examples/example-08-publish-to-others-services.md)
* Control the rendering step, like Publish a single format only as `quarto publish` will render all - see this [example](https://github.com/quarto-dev/quarto-actions/blob/main/examples/example-07-publish-single-format.md)

The latter is useful if you want to render and publish to one of the supported service - See  https://quarto.org/docs/publishing/#getting-started


Hope it helps
